test/solr_json_writer_test.rb

```diff
@@ -170,6 +170,27 @@ def logger_to_strio(strio)
     assert_length 1, @fake_http_client.post_args, "Has flushed to solr"
   end
 
+  it "defaults to not setting basic authentication" do
+    settings = { "solr.url" => "http://example.com/solr/foo" }
+    writer = Traject::SolrJsonWriter.new(settings)
+    auth = writer.instance_variable_get("@http_client")
+      .www_auth.basic_auth.instance_variable_get("@auth")
+    assert(auth.empty?)
```

Comment:
Thanks for the tests! I wish there was a way to test without all that reaching into instance variables, but there may not be a great one. Did you look at HTTPClient to see if there are any public API accessors that can return it's auth credentials?

Reply:
I could not figure out a way to do this without reaching in.  It's ugly but it's all I could figure out.